pkg/cloudprovider/providers/aws/aws_test.go

```diff
@@ -314,6 +324,23 @@ func (self *FakeMetadata) GetMetaData(key string) ([]byte, error) {
 		return []byte(self.aws.instanceId), nil
 	} else if key == "local-hostname" {
 		return []byte(self.aws.privateDnsName), nil
+	} else if key == "network/interfaces/macs/" {
+		macs := fmt.Sprintf(
+			"%s/\n%s/\n",
+			self.aws.networkInterfacesMac0,
+			self.aws.networkInterfacesMac1,
+		)
+		return []byte(macs), nil
+	} else if key == fmt.Sprintf(
+		"network/interfaces/macs/%s/vpc-id",
+		self.aws.networkInterfacesMac0,
```

Comment:
The formatting here makes this a little awkward.  If we can't just put it on one line, another option might be to check strings.HasPrefix("network/interfaces/macs/") and then inside the if block do a strings.Split on "/" and then handle the various cases (listing macs, fetching vpc-id for a mac).